src/miner.cpp

```diff
@@ -1286,6 +1278,56 @@ void StakeMiner(CWallet *pwallet)
             LogPrintf("WARN: StakeMiner: enablesidestaking was set in config but nothing has been allocated for distribution!");
     }
 
+    return fEnableSideStaking;
+}
+
+// This function parses the config file for the directives for stake splitting. It is used
+// in StakeMiner for the miner loop and also called by rpc getmininginfo.
+bool GetStakeSplitStatusAndParams(int64_t& nMinStakeSplitValue, double& dEfficiency)
+{
+    // Parse StakeSplit and SideStaking flags.
+    bool fEnableStakeSplit = GetBoolArg("-enablestakesplit");
+    LogPrintf("StakeMiner: fEnableStakeSplit = %u", fEnableStakeSplit);
+
+    // If stake output splitting is enabled, determine efficiency and minimum stake split value.
+    if (fEnableStakeSplit)
+    {
+        // Pull efficiency for UTXO staking from config, but constrain to the interval [0.75, 0.98]. Use default of 0.90.
+        dEfficiency = (double)GetArg("-stakingefficiency", 90) / 100;
+        if (dEfficiency > 0.98)
+            dEfficiency = 0.98;
+        else if (dEfficiency < 0.75)
+            dEfficiency = 0.75;
+
+        LogPrintf("StakeMiner: dEfficiency = %f", dEfficiency);
+
+        // Pull Minimum Post Stake UTXO Split Value from config or command line parameter.
+        // Default to 800 and do not allow it to be specified below 800 GRC.
+        nMinStakeSplitValue = max(GetArg("-minstakesplitvalue", MIN_STAKE_SPLIT_VALUE_GRC), MIN_STAKE_SPLIT_VALUE_GRC) * COIN;
+
+        LogPrintf("StakeMiner: nMinStakeSplitValue = %f", CoinToDouble(nMinStakeSplitValue));
+    }
+
+    return fEnableStakeSplit;
+}
+
+
+
+void StakeMiner(CWallet *pwallet)
+{
+    // Make this thread recognisable as the mining thread
+    RenameThread("grc-stake-miner");
+
+    int64_t nMinStakeSplitValue = 0;
+    double dEfficiency = 0;
+    SideStakeAlloc vSideStakeAlloc = {};
```

Comment:
You don't need to change this, but it's already initialized to empty on construction :)

Reply:
You are right. I forget sometimes.